docs/learn/computation/execution-model.md

```diff
@@ -12,12 +12,40 @@ When a transaction is executed successfully, it is included in the current block
 
 When block generation is complete, the block is propagated to all the other CNs. The other CNs all verify the propagated block and reach consensus on the verification results by using the BFT consensus algorithm. When the verification process completes successfully by the majority of CNs, the block is stored in the blockchain. Because the BFT consensus algorithm satisfies the immediate finality property, the block is final and will never be removed. After a block is finalized, the execution of all the transactions in that block are irreversibly guaranteed, and their execution results can be returned to the sender if requested.
 
+## Enhanced Randomness in Block Proposer Selection and Reward Distribution <a id="enhanced-randomness-in-block-proposer-selection-and-reward-distribution"></a>
+
+Kaia has implemented a new mechanism to introduce verifiable on-chain randomness in the block proposer selection and reward distribution processes. This mechanism involves two new fields in the block header: `randomReveal` and `mixHash`.
+
+In this system, validators generate and commit to random values. The `randomReveal` field in a block contains the revealed random value from a previous commitment. The `mixHash` is then computed using this revealed random value along with other block data, creating a source of randomness for the network.
+
+The block proposer selection process utilizes this generated randomness. The use of this randomness aims to make the selection process more unpredictable and fair, enhancing the overall security of the network.
+
+A key change in this mechanism is the separation of reward distribution from block creation. Rather than distributing rewards immediately upon block creation, the distribution is delayed. A separate transaction for reward distribution is included in a subsequent block. This transaction uses the randomness generated in the current block to determine reward distribution.
+
+The execution flow creates a cycle where each block's randomness influences future block proposer selections and reward distributions. This introduces an element of unpredictability to these processes while maintaining their verifiability.
```

Comment:
I think it's better to use `committee` instead of `reward distribution`. The reward distribution itself is calculated based on staking amounts, while validators who receive rewards are determined by randomness.

Reply:
And I think line#23 isn't a case of Klaytn. We don't execute transaction for reward distribution. The rewards are distributed at the end of block mining by modifying states directly. And I don't get the clear meaning of `separation of reward distribution from block creation`. If I missing something, please let me know. cc. @blukat29.

Reply:
@hyeonLewis I've replaced "reward distribution" with "committee selection" and wrote again the related parts. Please check on this and other remaining misconceptions if any.

Reply:
It looks clear to me now. Thanks.